.github/workflows/integration-tests.yaml

```diff
@@ -0,0 +1,212 @@
+          
+          # Configure profile using environment variables
+          cat > ~/.databrickscfg << EOF
+          [${{ matrix.profile }}]
+          host = $DATABRICKS_HOST
+          token = $DATABRICKS_TOKEN
+          EOF
+          
+          # Verify profile is configured correctly
+          databricks --profile ${{ matrix.profile }} current-user me
+
+      - name: Run Integration Tests
+        if: ${{ github.event_name == 'schedule' || github.event_name == 'push' || !github.event.inputs.cloud || github.event.inputs.cloud == 'both' || github.event.inputs.cloud == matrix.cloud }}
+        env:
+          # Test configuration
+          DATABRICKS_CONFIG_PROFILE: ${{ matrix.profile }}
+          DATABRICKS_CLOUD: ${{ matrix.cloud }}
+          DATABRICKS_HOST: ${{ secrets[matrix.host_var] }}
+          DATABRICKS_TOKEN: ${{ secrets[matrix.token_var] }}
+          # Use environment variables for catalog/schema names
+          TEST_CATALOG_NAME: ${{ vars.TEST_CATALOG_NAME || 'main_integration_tests' }}
+          TEST_SCHEMA_NAME: ${{ vars.TEST_SCHEMA_NAME || 'gha_integration_tests' }}
+          # GitHub-specific environment
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          # Run all tests including integration tests with retry on failures
+          pytest tests/ --integration --large -v \
+            --tb=short \
+            --reruns=2 \
+            --reruns-delay=30 \
+            --junit-xml=test-results-${{ matrix.cloud }}.xml \
+            --html=test-report-${{ matrix.cloud }}.html \
+            --self-contained-html
+
+      - name: Cleanup Credentials
+        if: always()  # Always run cleanup, even if tests fail
+        run: |
+          # Securely remove Databricks config files
+          shred -vfz -n 3 ~/.databrickscfg 2>/dev/null || rm -f ~/.databrickscfg
+          rm -rf ~/.databrickscfg.d
+          # Clear credential environment variables
+          unset DATABRICKS_HOST DATABRICKS_TOKEN
+          # Remove any temporary credential files
+          find /tmp -name "*databricks*" -type f -delete 2>/dev/null || true
+
+      - name: Upload Test Results
+        uses: actions/upload-artifact@v4
+        if: always()  # Upload even if tests failed
+        with:
+          name: test-results-${{ matrix.cloud }}
+          path: |
+            test-results-${{ matrix.cloud }}.xml
+            test-report-${{ matrix.cloud }}.html
+          retention-days: 30
+
+      - name: Upload Test Logs on Failure
+        uses: actions/upload-artifact@v4
+        if: failure()
+        with:
+          name: test-logs-${{ matrix.cloud }}
+          path: |
+            integration_test_*.log
+          retention-days: 7
+
+  notify-failure:
+    runs-on: 
+      group: databricks-field-eng-protected-runner-group
+      labels: [linux-ubuntu-latest]
+    needs: integration-tests
+    if: failure() && github.event_name == 'schedule'
+    steps:
+      - name: Send Email Notification
```

Comment:
I don't think we need email notification failures if we end up just running these on PRs, just leave a comment on the PR?